likecoin/js/sidebar/src/pages/LikeCoinPlugin.js

```diff
@@ -0,0 +1,289 @@
+        console.error(error);
+      }
+    }
+    async function onRegisterISCNSuccess(event) {
+      const { action, data } = JSON.parse(event.data);
+      const {
+        tx_hash: txHash, iscnId, iscnVersion, timestamp,
+      } = data;
+      if (action === 'ISCN_SUBMITTED') {
+        props.postISCNInfoData({
+          iscnHash: txHash,
+          iscnId,
+          iscnVersion,
+          timestamp,
+        });
+      }
+    }
+    // ---- Full flow. Article is not on Arweave net yet. Start LIKEPay
+    if (shouldStartProcess && parseFloat(LIKE) > 0) setShouldStartLIKEPay(true);
+    // ---- Direct register ISCN flow.
+    // If user has uploaded to Arweave before, then directly submit to ISCN
+    if (
+      shouldStartProcess
+      && parseFloat(LIKE) === 0
+      && arweaveIPFSHash
+      && arweaveIPFSHash.length > 1
+    ) {
+      setShouldDirectRegisterISCN(true);
+    }
+    // ---- Full flow step 1: LIKE Pay
+    if (
+      shouldStartProcess
+      && shouldStartLIKEPay
+      && (!arweaveIPFSHash || !arweaveId)
+      && memo
+      && siteurl
+      && title
+    ) {
+      const memoString = encodeURIComponent(memo);
+      const redirectString = encodeURIComponent(siteurl);
+      const titleString = encodeURIComponent(title);
+      const popUpWidget = `https://like.co/in/widget/wordpress?to=like-arweave&amount=${LIKE}&remarks=${memoString}&opener=1&redirect_uri=${redirectString}&title=${titleString}`;
+      setPopUpWindow(
+        window.open(
+          popUpWidget,
+          'likePayWindow',
+          'menubar=no,location=no,width=576,height=768',
+        ),
+      );
+      setShouldStartProcess(false);
+      setShouldStartLIKEPay(false);
+      setShouldContinueUploadArweave(true);
+      window.addEventListener(
+        'message',
+        (event) => onLIKEPaySuccessUploadToArweave(event),
+        false,
+      );
+      // ---- Direct register ISCN flow.
+    } else if (
+      shouldStartProcess
+      && shouldDirectRegisterISCN
+      && arweaveIPFSHash
+      && arweaveId
+      && memo
+      && siteurl
+      && title
+    ) {
+      const tagsArray = tags || [];
+      const tagsString = tagsArray.join(',');
+      if (url && memo && title && fingerprints.length > 0) {
+        const urlString = encodeURIComponent(url);
+        const memoString = encodeURIComponent(memo);
+        const redirectString = encodeURIComponent(siteurl);
+        const titleString = encodeURIComponent(title);
+        const fingerprint = fingerprints.join(',');
+        const popUpWidget = `https://like.co/in/widget/wordpress?fingerprint=${fingerprint}&publisher=&title=${titleString}&tags=${tagsString}&url=${urlString}&to=like-arweave&amount=0&remarks=${memoString}&opener=1&redirect_uri=${redirectString}`;
+
+        setPopUpWindow(
+          window.open(
+            popUpWidget,
+            'likePayWindow',
+            'menubar=no,location=no,width=576,height=768',
+          ),
+        );
```

Comment:
This part can also be wrapped into a function, with `window.addEventListener('message', onRegisterISCNSuccess, false);`